classes/autoptimizeStyles.php

```diff
@@ -343,15 +343,20 @@ public function read( $options )
                                 if ( ! empty( $minified_url ) ) {
                                     // Replace orig URL with cached minified URL.
                                     $new_tag = str_replace( $url, $minified_url, $tag );
+                                } else {
+                                    // Remove the original style tag, because cache content is empty.
+                                    $new_tag = '';
                                 }
                             }
                         }
 
-                        // Optionally defer (preload) non-aggregated CSS.
-                        $new_tag = $this->optionally_defer_excluded( $new_tag, $url );
+                        if ( '' !== $new_tag) {
+                            // Optionally defer (preload) non-aggregated CSS.
+                            $new_tag = $this->optionally_defer_excluded( $new_tag, $url );
+                        }
 
                         // And replace!
-                        if ( '' !== $new_tag && $new_tag !== $tag ) {
```

Comment:
I've been thinking about this some more while working in the garden; I'm kind of wary of removing part of the HTML just like that to be honest.

To be clear; I'm sure that in most scenario's this would cause no issues, but I have seen too many edge cases where things broke even if it did not make sense. This is why generally speaking I try to avoid potentially breaking changes by putting them behind a filter (or an option, but that's for more visible functionality) that is default off, allowing people that need the new functionality to enable it, but being conscious of it.

So in this case I'd prefer the condition to become;
```php
if ( ( '' !== $new_tag && $new_tag !== $tag ) || ( '' === $new_tag && apply_filters( 'autoptimize_filter_css_remove_empty_files', false ) ) ) {
```

Maybe one day we can make it default true, but we need to be careful when pushing new code to nearly 1,5 million active installs ;-)

Reply:
Changed